File: apps/java-sdk/src/main/java/com/firecrawl/models/Monitor.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Monitor {
    private String id;
    private String name;
    private String status;
    private MonitorSchedule schedule;
    private String nextRunAt;
    private String lastRunAt;
    private String currentCheckId;
    private List<Map<String, Object>> targets;
    private Map<String, Object> webhook;
    private Map<String, Object> notification;
    private int retentionDays;
    private Integer estimatedCreditsPerMonth;
    private MonitorSummary lastCheckSummary;
    private String goal;
    private Boolean judgeEnabled;
    private String createdAt;
    private String updatedAt;

    public String getId() { return id; }
    public String getName() { return name; }
    public String getStatus() { return status; }
    public MonitorSchedule getSchedule() { return schedule; }
    public String getNextRunAt() { return nextRunAt; }
    public String getLastRunAt() { return lastRunAt; }
    public String getCurrentCheckId() { return currentCheckId; }
    public List<Map<String, Object>> getTargets() { return targets; }
    public Map<String, Object> getWebhook() { return webhook; }
    public Map<String, Object> getNotification() { return notification; }
    public int getRetentionDays() { return retentionDays; }
    public Integer getEstimatedCreditsPerMonth() { return estimatedCreditsPerMonth; }
    public MonitorSummary getLastCheckSummary() { return lastCheckSummary; }
    public String getGoal() { return goal; }
    public Boolean getJudgeEnabled() { return judgeEnabled; }
    public String getCreatedAt() { return createdAt; }
    public String getUpdatedAt() { return updatedAt; }
}

File: apps/php-sdk/src/Models/Monitor.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class Monitor
{
    /**
     * @param list<array<string, mixed>> $targets Each target array has a
     *     `type` of `scrape`, `crawl`, or `search`, plus an optional
     *     `id`. `scrape`/`crawl` targets carry `urls`/`url` and
     *     `scrapeOptions`/`crawlOptions`. `search` targets carry
     *     `queries` (list<string>, required) and optional
     *     `searchWindow` (one of `5m`, `15m`, `1h`, `6h`, `24h`, `7d`),
     *     `includeDomains` (list<string>), `excludeDomains`
     *     (list<string>), and `maxResults` (int). All keys are camelCase.
     * @param array<string, mixed>|null  $schedule
     * @param array<string, mixed>|null  $webhook
     * @param array<string, mixed>|null  $notification
     * @param array<string, mixed>|null  $lastCheckSummary
     */
    public function __construct(
        private readonly ?string $id = null,
        private readonly ?string $name = null,
        private readonly ?string $status = null,
        private readonly ?array $schedule = null,
        private readonly ?string $nextRunAt = null,
        private readonly ?string $lastRunAt = null,
        private readonly ?string $currentCheckId = null,
        private readonly array $targets = [],
        private readonly ?array $webhook = null,
        private readonly ?array $notification = null,
        private readonly ?int $retentionDays = null,
        private readonly ?int $estimatedCreditsPerMonth = null,
        private readonly ?array $lastCheckSummary = null,
        private readonly ?string $goal = null,
        private readonly bool $judgeEnabled = false,
        private readonly ?string $createdAt = null,
        private readonly ?string $updatedAt = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            id: isset($data['id']) ? (string) $data['id'] : null,
            name: isset($data['name']) ? (string) $data['name'] : null,
            status: isset($data['status']) ? (string) $data['status'] : null,
            schedule: isset($data['schedule']) && is_array($data['schedule']) ? $data['schedule'] : null,
            nextRunAt: isset($data['nextRunAt']) ? (string) $data['nextRunAt'] : null,
            lastRunAt: isset($data['lastRunAt']) ? (string) $data['lastRunAt'] : null,
            currentCheckId: isset($data['currentCheckId']) ? (string) $data['currentCheckId'] : null,
            targets: isset($data['targets']) && is_array($data['targets']) ? $data['targets'] : [],
            webhook: isset($data['webhook']) && is_array($data['webhook']) ? $data['webhook'] : null,
            notification: isset($data['notification']) && is_array($data['notification']) ? $data['notification'] : null,
            retentionDays: isset($data['retentionDays']) ? (int) $data['retentionDays'] : null,
            estimatedCreditsPerMonth: isset($data['estimatedCreditsPerMonth']) ? (int) $data['estimatedCreditsPerMonth'] : null,
            lastCheckSummary: isset($data['lastCheckSummary']) && is_array($data['lastCheckSummary']) ? $data['lastCheckSummary'] : null,
            goal: isset($data['goal']) ? (string) $data['goal'] : null,
            judgeEnabled: isset($data['judgeEnabled']) ? (bool) $data['judgeEnabled'] : false,
            createdAt: isset($data['createdAt']) ? (string) $data['createdAt'] : null,
            updatedAt: isset($data['updatedAt']) ? (string) $data['updatedAt'] : null,
        );
    }

    public function getId(): ?string { return $this->id; }
    public function getName(): ?string { return $this->name; }
    public function getStatus(): ?string { return $this->status; }
    /** @return array<string, mixed>|null */
    public function getSchedule(): ?array { return $this->schedule; }
    public function getNextRunAt(): ?string { return $this->nextRunAt; }
    public function getLastRunAt(): ?string { return $this->lastRunAt; }
    public function getCurrentCheckId(): ?string { return $this->currentCheckId; }
    /** @return list<array<string, mixed>> */
    public function getTargets(): array { return $this->targets; }
    /** @return array<string, mixed>|null */
    public function getWebhook(): ?array { return $this->webhook; }
    /** @return array<string, mixed>|null */
    public function getNotification(): ?array { return $this->notification; }
    public function getRetentionDays(): ?int { return $this->retentionDays; }
    public function getEstimatedCreditsPerMonth(): ?int { return $this->estimatedCreditsPerMonth; }
    /** @return array<string, mixed>|null */
    public function getLastCheckSummary(): ?array { return $this->lastCheckSummary; }
    public function getGoal(): ?string { return $this->goal; }
    public function getJudgeEnabled(): bool { return $this->judgeEnabled; }
    public function getCreatedAt(): ?string { return $this->createdAt; }
    public function getUpdatedAt(): ?string { return $this->updatedAt; }
}
